pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -185,48 +221,38 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 	// Start deploy klusterlet components
 	// Ensure the existence namespaces for klusterlet and klusterlet addon
 	// Sync pull secret to each namespace
-	namespaces := []string{config.KlusterletNamespace, fmt.Sprintf("%s-addon", config.KlusterletNamespace)}
-	for _, namespace := range namespaces {
-		err := n.ensureNamespace(ctx, klusterlet.Name, namespace)
-		if err != nil {
-			return err
-		}
-
-		// Sync pull secret
-		_, _, err = resourceapply.SyncSecret(
-			n.kubeClient.CoreV1(),
-			controllerContext.Recorder(),
-			n.operatorNamespace,
-			imagePullSecret,
-			namespace,
-			imagePullSecret,
-			[]metav1.OwnerReference{},
-		)
+	err = n.syncPullSecret(ctx, n.kubeClient, n.kubeClient, klusterlet.Name, config.KlusterletNamespace, controllerContext.Recorder())
+	if err != nil {
+		return err
+	}
+	// For now, whether in Default or Detached mode, the addons will be deployed on the managed cluster.
+	// sync image pull secret from management cluster to managed cluster for addon namespace
+	// TODO(zhujian7): In the future, we may consider deploy addons on the management cluster in Detached mode.
+	addonNamespace := fmt.Sprintf("%s-addon", config.KlusterletNamespace)
+	err = n.syncPullSecret(ctx, n.kubeClient, managedClusterClients.kubeClient, klusterlet.Name, addonNamespace, controllerContext.Recorder())
+	if err != nil {
+		return err
+	}
 
+	if detachedMode {
```

Comment:
this is confused..do we really need to apply any role/rolebinding in this ns?

Reply:
need to create service account in this ns.

Reply:
need to create service account in this ns